src/framework/auth/services/token/token.ts

```diff
@@ -36,8 +38,26 @@ export class NbAuthSimpleToken extends NbAuthToken {
   static NAME = 'nb:auth:simple:token';
 
   constructor(protected readonly token: any,
-              protected readonly ownerStrategyName: string) {
+
+              protected readonly ownerStrategyName: string,
+              protected createdAt?: Date) {
     super();
+    if (!this.createdAt) {
```

Comment:
probably we should go a bit different way?

```
this.createdAt = this.prepareCreatedAt(createdAt);
```
and then 

```
// this is for simple token
protected prepareCreatedAt(date: Date) {
  return date;
}

// this is for jwt
protected prepareCreatedAt(date: Date) {
  const decoded = this.getPayload();
  return decoded && decoded.ait ? new Date(Number(decoded.iat) * 1000) : date;
}
```

Reply:
@nnixaa can you confirm what you mean : 
- `createdAt` is an optional parameter of the constructor
- if present, we use it to initialize the simpleToken `createdAt` attribute 
- if present, we use it to initialize the "without iat AuthToken " `createdAt` attribute 
- if not present, i think we have to use `new Date()` by default.

So i guess the code should be :
    `this.createdAt = this.prepareCreatedAt(createdAt);`

and then 
```
protected prepareCreatedAt(date: Date) {
    // For simple tokens, if not set the creation date is 'now'
    return date ? date : new Date();
  }
```
and for NbAuthJWTToken
```
  protected prepareCreatedAt(date: Date) {
    let decoded = null;
    try { // needed as getPayload() throws error and we want the token to be created in any case
      decoded = this.getPayload();
    }
    finally {
      return decoded && decoded.iat ? new Date(Number(decoded.iat) * 1000) : (date ? date : new Date());

    }
  }
```

What do you think ?

Reply:
Yes this is correct, with the only code style change I left.